### PR TITLE
Add Job Viewer visual element to display job details

### PR DESCRIPTION
**Pull request by Nupoor10**
This PR is regarding issue #412 

It includes addition of a new job viewer visual element that is similar to the details panel of the Job Selector visual element
1. Added a new React component JobDetailsViewer
2. Displayed all necessary information outlined in the issue with the correct conditions
3. Added a new method to fetch complete job details
4. Added a new method to check if a job can be cancelled
5. Modified the _GuiCoreLib.py and viselements.json files appropriately

### CODE REVIEW

**Comment by Nupoor10** [comment]
@quest-bot loot #412

**Comment by Nupoor10** [comment]
I had a few questions regarding this PR.
1. In which subfolder should I add the test file for the JobDetailsViewer component?
2. I tried to implement this component but I get a job_viewer not registered message so how can I implement it?
3. I have implemented the is_cancellable method in job.py folder that returns a boolean value, is that the expected implementation or should it return a ReasonCollection?
4. In the _GuiCoreLib.py file I Have used the __JOB_SELECTOR_ERROR_VAR variable so is it correct?

Additionally if there are any changes do let me know.

File: frontend/taipy/src/JobDetailsViewer.tsx
Comment (FredLL-Avaiga) [inline frontend/taipy/src/JobDetailsViewer.tsx:1]: Can you not reuse the JobViewer component ?

**Review by FredLL-Avaiga — CHANGES_REQUESTED** [review]
re-use :-)

**Comment by FredLL-Avaiga** [comment]
> 1. In which subfolder should I add the test file for the JobDetailsViewer component?

in the same folder, the file name should end with spec.tsx (check `ScenarioSelector.spec.tsx`)

> 2. I tried to implement this component but I get a job_viewer not registered message so how can I implement it?

frontend component name is calculated from backend name if not provided
you need to add the `react_component` parameter to your `Element` instantiation

> 3. I have implemented the is_cancellable method in job.py folder that returns a boolean value, is that the expected 
implementation or should it return a ReasonCollection?

a `ReasonCollection` would be nicer and more consistent

> 4. In the _GuiCoreLib.py file I Have used the __JOB_SELECTOR_ERROR_VAR variable so is it correct?

No, you need to define a specific error variable name for this new component

PS: please fix the linter errors and update your branch

**Comment by FredLL-Avaiga** [comment]
Are you still working on this PR @Nupoor10 ?

**Comment by FredLL-Avaiga** [comment]
> Are you still working on this PR @Nupoor10 ?

If you're not, please close this PR

**Pull request closed without merging**